### PR TITLE
Optimize memory usage when stream flush and save

### PR DESCRIPTION
**Pull request by snowcrumble**
# PR Details

Optimize memory usage when stream flush and save

## Description

1. Record streams in File when new a stream
2. Do not close temp file when flush stream
3. use io.Copy to copy from stream buffer reader into zip writer.

## Related Issue

#650 

## Motivation and Context

Avoid instantaneous memory usage.

## How Has This Been Tested

I fix a OOM problem in k8s.

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by xuri** [comment]
Thanks for your pull request. Please make sure to pass the go test.

File: file.go [hunk 1]
```diff
@@ -18,6 +18,8 @@ import (
 	"fmt"
 	"io"
 	"os"
+
+	"gitlab.com/laiye-backend-repos/go-utils/xzap"
```
Comment (xuri) [inline file.go:22]: Please don't use private packages here

File: go.sum [hunk 2]
```diff
@@ -1,7 +1,26 @@
+cloud.google.com/go v0.26.0/go.mod h1:aQUYkXzVsufM+DwF1aE+0xfcU+56JwCaLick0ClmMTw=
+github.com/BurntSushi/toml v0.3.1/go.mod h1:xHWCNGjB5oqiDr8zfno3MHue2Ht5sIBksp03qcyfWMU=
```
Comment (xuri) [inline go.sum:2]: Please remove unnecessary packages in `go.sum`.

**Comment by Paxa** [comment]
Can you give usage example please?

btw this change will be very helpful for me

**Review by xuri — APPROVED** [review]

**Pull request merged**